src/main/java/jenkins/plugins/git/GitSCMSource.java

```diff
@@ -287,43 +295,97 @@ public ListBoxModel doFillGitToolItems() {
             }
             SecurityContext old = jenkins.getACL().impersonate(ACL.SYSTEM);
             try {
-                for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
-                    for (SCMSource source : owner.getSCMSources()) {
-                        if (source instanceof GitSCMSource) {
-                            GitSCMSource git = (GitSCMSource) source;
-                            if (git.ignoreOnPushNotifications) {
-                              continue;
+                if (branches.length > 0) {
+                    final URIish u = uri;
+                    for (final String branch: branches) {
+                        SCMHeadEvent.fireNow(new SCMHeadEvent<String>(SCMEvent.Type.UPDATED, branch){
+                            @Override
+                            public boolean isMatch(@NonNull SCMNavigator navigator) {
+                                return false;
                             }
-                            URIish remote;
-                            try {
-                                remote = new URIish(git.getRemote());
-                            } catch (URISyntaxException e) {
-                                // ignore
-                                continue;
+
+                            @NonNull
+                            @Override
+                            public String getSourceName() {
+                                return null;
```

Comment:
The NonNull annotation conflicts with the `return null` statement.

Should the method return at empty string rather than null, should the annotation be removed, or should an exception be thrown to indicate this code was never expected to be called?